File: src/output/read_rdf.m
% Used to plot MDMC generated rdf functions

function read_rdf(filename, format)

s = xmlread(filename);

mylist = s.getElementsByTagName('g-of-r');

r = zeros([1 mylist.getLength()]);
g = zeros([1 mylist.getLength()]);

for i = 0 : mylist.getLength()-1
  mylist.item(i).getTagName();
  r(i+1) = str2num(mylist.item(i).getAttribute('r'));
  g(i+1) = str2num(mylist.item(i).getAttribute('g'));
end 

plot(r, g, format)
xlabel('r[0.1nm]')
ylabel('g(r)')

File: src/output/read_rdf.py
"""
Used to plot MDMC generated rdf functions
"""

import xml.etree.ElementTree as ET
import numpy as np
import matplotlib.pyplot as plt


def read_rdf(filename, format_style='b-'):
    """Read and plot MDMC generated rdf functions."""
    
    tree = ET.parse(filename)
    root = tree.getroot()
    
    elements = root.findall('g-of-r')
    
    r = np.zeros(len(elements))
    g = np.zeros(len(elements))
    
    for i, elem in enumerate(elements):
        r[i] = float(elem.get('r'))
        g[i] = float(elem.get('g'))
    
    plt.plot(r, g, format_style)
    plt.xlabel('r[0.1nm]')
    plt.ylabel('g(r)')
    plt.show()
    
    return {'r': r, 'g': g}


if __name__ == '__main__':
    import sys
    if len(sys.argv) > 1:
        format_style = sys.argv[2] if len(sys.argv) > 2 else 'b-'
        read_rdf(sys.argv[1], format_style)
